src/cosmos/engine/CosmosEngine.ts

```diff
@@ -1142,8 +1268,129 @@ export class CosmosEngine extends CurrencyEngine<
       edgeTransaction.otherParams
     )
     const keys = asCosmosPrivateKeys(this.currencyInfo.pluginId)(privateKeys)
-    const txRawBytes = base16.parse(unsignedTxHex)
-    const { bodyBytes } = TxRaw.decode(txRawBytes)
+
+    let bodyBytes: Uint8Array
+
+    // Check if this is cosmos DEX swap transaction data or traditional hex
+    if (unsignedTxHex.startsWith('{')) {
```

Comment:
This seems fragile and is very hard to understand. Can you reuse any of the work in signMessage to simplify this?